codes/typescript/tsconfig.json

```diff
@@ -1,8 +1,31 @@
 {
     "compilerOptions": {
```

Comment:
I think the config is too complex. Could we keep some of them default? 

(This book is not for TypeScript professionalism. We want to keep it as simple as possible.)